robottelo/utils/__init__.py

```diff
@@ -80,3 +91,178 @@ def slugify_component(string, keep_hyphens=True):
+            consumer_export.seek(0)
+            signature = self.private_key.sign(
+                consumer_export.read(), padding.PKCS1v15(), hashes.SHA256()
+            )
+            manifest_zip.writestr('signature', signature)
+        # Make sure that the file-like object is at the beginning and
+        # ready to be read.
+        manifest.seek(0)
+        return manifest
+
+    def original(self, name='default'):
+        """Returns the original manifest as a file-like object.
+
+        :param name: A name of the manifest as defined in robottelo.properties
+
+        Be aware that using the original manifest and not removing it
+        afterwards will make it impossible to import it to any other
+        Organization.
+
+        Make sure to close the returned file-like object in order to clean up
+        the memory used to store it.
+        """
+        if self.signing_key is None or self.template is None or self.template.get(name) is None:
+            self._download_manifest_info(name)
+        return io.BytesIO(self.template[name])
+
+
+# Cache the ManifestCloner in order to avoid downloading the manifest template
+# every single time.
+_manifest_cloner = ManifestCloner()
+
+
+class Manifest:
+    """Class that holds the contents of a manifest with a generated filename
+    based on ``time.time``.
+
+    To ensure that the manifest content is closed use this class as a context
+    manager with the ``with`` statement::
+
+        with Manifest() as manifest:
+            # my fancy stuff
+    """
+
+    def __init__(self, content=None, filename=None, org_environment_access=False, name='default'):
+        self._content = content
+        self.filename = filename
+
+        if self._content is None:
+            self._content = _manifest_cloner.clone(
+                org_environment_access=org_environment_access, name=name
+            )
+        if self.filename is None:
+            self.filename = f'/var/tmp/manifest-{int(time.time())}.zip'
+
+    @property
+    def content(self):
+        if not self._content.closed:
+            # Make sure that the content is always ready to read
+            self._content.seek(0)
+        return self._content
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, type, value, traceback):
+        if not self.content.closed:
+            self.content.close()
+
+
+def clone(org_environment_access=False, name='default'):
+    """Clone the cached manifest and return a ``Manifest`` object.
+
+    :param org_environment_access: Whether to modify consumer content
+        access mode to org_environment (Golden ticket enabled manifest).
+
+    :param name: key name of the fake_manifests.url dict defined in
+        robottelo.properties
+
+    Is hightly recommended to use this with the ``with`` statement to make that
+    the content of the manifest (file-like object) is closed properly::
+
+        with clone() as manifest:
+            # my fancy stuff
```

Comment:
```suggestion
           
```

Reply:
@adarshdubey-star It is done as an example to use  it.